geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/KafkaFeatureCache.scala

```diff
@@ -50,7 +55,35 @@ object KafkaFeatureCache {
             }
           }
         }
-        builder.expireAfterWrite(expiry.toMillis, TimeUnit.MILLISECONDS).removalListener(listener)
+        if(eventTimeExpiry) {
+          builder.expireAfter(new Expiry[String, T] {
+            private def getExpirationTime(value: T, curTimeNanos: Long) = {
+              val eventTime = getEventTime(value)
+              if (eventTime == null) {
+                logger.warn("Unexpected null event time, expiring immediately")
+                0L
+              } else {
+                val dtg = Instant.ofEpochMilli(eventTime.getTime)
+                val now = Instant.ofEpochMilli(TimeUnit.NANOSECONDS.toMillis(curTimeNanos))
```

Comment:
think it would make a performance difference to just compare them as longs? doesn't seem like we're gaining much from using instants instead